src/pkgcheck/checks/metadata.py

```diff
@@ -1044,3 +1044,85 @@ def feed(self, pkg):
             deprecated = set(
                 x for x in bad if x.startswith("no") and x[2:] in self.known_restricts)
             yield BadRestricts(pkg, bad.difference(deprecated), deprecated)
+
+
+class MissingUnpackerDep(base.Warning):
+    """Package is missing dependency on unpacker.
+
+    Package is using an archive format for which unpacker is not provided
+    by the system set, and lacks explicit dependency on the unpacker. """
+
+    __slots__ = ("category", "package", "version", "formats", "unpackers")
+    threshold = base.versioned_feed
+
+    def __init__(self, pkg, formats, unpackers):
+        super().__init__()
+        self._store_cpv(pkg)
+        self.formats = tuple(sorted(formats))
+        self.unpackers = tuple(sorted(unpackers))
+
+    @property
+    def short_desc(self):
+        return (f"Package is using file formats: {', '.join(self.formats)} "
+                f"in SRC_URI but is missing bdepend on unpackers: "
+                f"{' '.join(str(x) for x in self.unpackers)}")
+
+
+class MissingUnpackerDepCheck(base.Template):
+    """Check whether package is missing unpacker dependencies."""
+
+    feed_type = base.versioned_feed
+
+    known_results = (MissingUnpackerDep,)
+    required_addons = (addons.UseAddon,)
+
+    unpackers = {
```

Comment:
I generally try to keep class attrs immutable unless otherwise required, i.e. use an ImmutableDict here.